### ISSUE TITLE
Setting custom `accel_profile` on touchad causes workspace swipe gesture to crash and move workspace focus erratically

### ISSUE THREAD

**Issue by m-bartlett** [hyprwm/Hyprland#8900]
### Already reported ? *

- [X] I have searched the existing open and closed issues.

### Regression?

Yes

### System Info and Version

<details> <summary>System/Version info</summary>

```sh
Hyprland 0.46.0 built from branch  at commit 788ae588979c2a1ff8a660f16e3c502ef5796755 dirty (version: bump to 0.46.0).
Date: Mon Dec 16 16:38:56 2024
Tag: v0.46.0, commits: 1
built against:
 aquamarine 0.5.1
 hyprlang 0.6.0
 hyprutils 0.3.0
 hyprcursor 0.1.11
 hyprgraphics 0.1.1


flags set:
debug


System Information:
System name: Linux
Node name: magnum
Release: 6.12.7-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Fri, 27 Dec 2024 14:24:37 +0000


GPU information: 
00:02.0 VGA compatible controller [0300]: Intel Corporation TigerLake-H GT1 [UHD Graphics] [8086:9a60] (rev 01) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
  hy3 by outfoxxed ver 0.1

======Config-Start======


Config File: /home/m/.config/hypr/hyprland.conf: Read Succeeded
# -*- mode: Makefile -*-

debug {
    disable_logs = false
}

# Variables {{{
# https://wiki.hyprland.org/Configuring/Keywords/#defining-variables
$HYPER = MOD3
#localectl list-x11-keymap-options
#xkbcli list
# SHIFT CAPS CTRL/CONTROL ALT MOD2 MOD3 SUPER/WIN/LOGO/MOD4 MOD5

# source = ~/.config/hypr/myColors.conf

env = XDG_CURRENT_DESKTOP,Hyprland
env = XDG_SESSION_TYPE,wayland
env = XDG_SESSION_DESKTOP,Hyprland
env = ADW_DISABLE_PORTA,1
env = QT_AUTO_SCREEN_SCALE_FACTOR,1
env = QT_QPA_PLATFORM,wayland;xcb
env = QT_WAYLAND_DISABLE_WINDOWDECORATION,1
env = QT_QPA_PLATFORMTHEME,qt5ct
env = GDK_BACKEND,wayland,x11
env = SDL_VIDEODRIVER,wayland
env = CLUTTER_BACKEND,wayland
env = GTK_IM_MODULE,fcitx
env = QT_IM_MODULE,fcitx
env = XMODIFIERS,@im,fcitx
env = SDL_IM_MODULE,fcitx
env = GLFW_IM_MODULE,ibus

# }}}

# Input device configuration {{{

general {
#     no_cursor_warps = true
    no_focus_fallback = true
}

cursor {
    hotspot_padding = 0
    inactive_timeout = 10
    no_warps = true
    zoom_factor = 1.0
    zoom_rigid = false
    hide_on_key_press= true
}

input {
    kb_variant =
    kb_model =
    kb_layout = bettercaps

    touchpad {
        natural_scroll = no
        disable_while_typing = true
        scroll_factor = 0.33
    }
    # force_no_accel, follow_mouse, float_switch_override_focus, scroll_factor

    float_switch_override_focus = 0 # don't focus floating windows on hover

    # 0 - Cursor movement will not change focus.
    # 1 - Cursor movement will always change focus to the window under the cursor.
    # 2 - Cursor focus will be detached from keyboard focus. Clicking on a window will move keyboard focus to that window.
    # 3 - Cursor focus will be completely separate from keyboard focus. Clicking on a window will not change keyboard focus.
    follow_mouse = 2

    # 0 = focux next window, 1 = focus window under cursor.
    focus_on_close = 1
}

device {
    name = mx-anywhere-2s-mouse
    sensitivity = 1
    accel_profile = flat
    sensitivity = 0.66 # -1.0 - 1.0, 0 means no modification.
}

device {
    name = dll0945:00-04f3:311c-touchpad
    sensitivity = 1.0 # -1.0 - 1.0, 0 means no modification.
    accel_profile = custom 0.9, 0.0, 1.0
    #accel_profile = flat
}

device {
    name = apple-inc.-magic-trackpad-1
    sensitivity = 1 # -1.0 - 1.0, 0 means no modification.
    #accel_profile = custom 0.95, 0.0, 1.0
    disable_while_typing = false
    accel_profile = flat
}

device {
    name = apple-inc.-magic-trackpad
    sensitivity = 1 # -1.0 - 1.0, 0 means no modification.
    #accel_profile = custom 0.95, 0.0, 1.0
    disable_while_typing = false
    accel_profile = flat
}

gestures {
    workspace_swipe = false
    workspace_swipe_create_new = false
    workspace_swipe_use_r = true
    #workspace_swipe_fingers = 4
    workspace_swipe_min_fingers = 4
    workspace_swipe_distance = 3000
    workspace_swipe_invert = false
    #workspace_swipe_min_speed_to_force = 0
    #workspace_swipe_cancel_ratio = 0.4
}

binds {
    scroll_event_delay = 100
    focus_preferred_method = 0
}

# }}}

# Display, Decorations, and Animations {{{

monitor=,1920x1200, 0x0,1

$gaps_in_default = 5
$gaps_out_default = 10
$border_size_default = 1

general {
    gaps_in = $gaps_in_default
    gaps_out = $gaps_out_default
    gaps_workspaces = 50
    border_size = $border_size_default
    col.active_border = rgba(1793D1c0)
    col.inactive_border = rgba(59595933)
}

decoration {
    rounding = 3
    inactive_opacity = 0.87
    # screen_shader = /path/to/shader.frag
    dim_special = 0.33

    blur {
        enabled = yes
        size = 4
        passes = 2
        new_optimizations = true
        ignore_opacity = true
        special = false
        vibrancy = 0.5
            # xray = floating layers ignore windows underneath
        xray = no
    }

    shadow {
        enabled = true
        range = 20
        render_power = 3
        sharp = false
        ignore_window = true
        color = rgba(1a1a1a88)
        color_inactive = rgba(1a1a1a66)
        offset = 0 3
        scale = 1.0
    }
}

animations {
    # Some default animations, see https://wiki.hyprland.org/Configuring/Animations/ for more
    enabled = yes

    ### animation = NAME, ONOFF, DURATION in deciseconds, CURVE [,STYLE]

    bezier = overshootBounce, 0.05, 0.9, 0, 1

    animation = windows, 1, 7, overshootBounce
    animation = windowsIn,  1, 5, default, popin 50%
    animation = windowsOut, 1, 100, default, popin
    animation = border, 1, 10, default
    animation = fade, 1, 7, default
    animation = workspaces, 1, 8, default
    animation = specialWorkspace, 1, 6, default, slidevert

    animation = layers, 1, 3, default, popin 80%
}

misc {
    disable_hyprland_logo = true
    animate_manual_resizes = true
    vfr = true
    close_special_on_empty = true
    always_follow_on_dnd = true
    # 0 - behind, 1 - takes over, 2 - unfullscreen the current fullscreen window
    new_window_takes_over_fullscreen = 2
#     cursor_zoom_factor = 1.0
    focus_on_activate = true
}

xwayland {
    use_nearest_neighbor = true
    force_zero_scaling = true
}

# }}}

# Layer & Application Specific Configuration {{{
# See https://wiki.hyprland.org/Configuring/Window-Rules/ for more

layerrule = blur,launcher
layerrule = noanim,background
layerrule = noanim,selection

# generic
windowrulev2 = bordercolor rgb(1793D1) rgb(888888), floating:1
windowrulev2 = animation slide,floating:1
windowrulev2 = noshadow,floating:0 focus:0

#waybar
layerrule = blur,waybar
layerrule = ignorealpha 0.5,waybar

#sway-notification-center
layerrule = blur,swaync-control-center
layerrule = ignorezero,swaync-control-center
layerrule = animation slide right,swaync-control-center
layerrule = blur,swaync-notification-window
layerrule = ignorealpha 0.25,swaync-notification-window
layerrule = animation slide right,swaync-notification-window

#swayosd
layerrule = blur,swayosd
layerrule = ignorezero,swayosd

# Drop-in terminal
$TERMINAL_SUMMON_WORKSPACE = hyprterm
workspace = special:$TERMINAL_SUMMON_WORKSPACE, gapsin:15, gapsout:15, bordersize:3
bind = SUPER, return, togglespecialworkspace, $TERMINAL_SUMMON_WORKSPACE
bind = SUPER CTRL, return, movetoworkspace, special:$TERMINAL_SUMMON_WORKSPACE
windowrulev2 = float,onworkspace:$TERMINAL_SUMMON_WORKSPACE

# codium
windowrule = float,title:Open File
windowrule = float,title:Open Folder
windowrule = float,title:Add Folder to Workspace
windowrule = float,title:Create File

# Float by default
windowrule = float,org.manjaro.pamac.manager
windowrule = float,^(xdg-desktop-portal.*)$
#windowrule = float,^(engrampa)$,title:^(Extract)$
windowrule = float,^(engrampa)$
windowrule = float,^(Vita3K)$
windowrule = float,^(blueman-manager)$
windowrule = float,^(pavucontrol)$
windowrule = float,^(nwg-look)$
windowrule = float, title:File Operation Progress
windowrulev2 = float, title:File Operation Progress
windowrule = float,title:Confirm to replace files
windowrule = float,calculator
windowrule = float,gnome-system-monitor
windowrule = float,org.gnome.SystemMonitor
windowrule = float,org.pulseaudio.pavucontrol
windowrule = float,gui-confirm

# mpv
windowrule = suppressevent maximize fullscreen,^(mpv)$

# freecad performance
windowrule = noblur,^(FreeCAD)$
windowrule = noanim,^(FreeCAD)$

# ksnip
windowrule = float,^(org.ksnip.ksnip)$
windowrule = minsize 400 600,^(org.ksnip.ksnip)$
windowrule = center,^(org.ksnip.ksnip)$

# Remove effects on XWayland windows to improve rendering speed
windowrulev2 = noblur,xwayland:1
windowrulev2 = noanim,xwayland:1
windowrulev2 = forcergbx,xwayland:1

# gsimplecal
windowrule = move onscreen cursor -50% -10%,^(gsimplecal)$

# wofi
windowrule = animation slide,wofi
windowrule = noborder,wofi
layerrule = blur,wofi
layerrule = animation popin,wofi
# layerrule = noanim,wofi

# Popup text input box for keybinds
windowrule = float,gui-text-input
windowrule = animation slide,gui-text-input

# smile (emoji picker)
windowrule = move onscreen cursor -50% -10%,it.mijorus.smile
windowrule = float,it.mijorus.smile
windowrule = animation slide,it.mijorus.smile

# flameshot
windowrule = noanim,flameshot
windowrule = noblur,flameshot
windowrule = noborder,flameshot
windowrule = float,flameshot
windowrule = noanim,slurp

# krita
windowrule = focusonactivate on,krita
# windowrule = stayfocused, krita

# }}}

# Layout & Plugins {{{

general {
    layout = hy3
}

dwindle {
    # See https://wiki.hyprland.org/Configuring/Dwindle-Layout/ for more
    pseudotile = yes
    preserve_split = yes # you probably want this
        ## https://wiki.hyprland.org/Configuring/Workspace-Rules/#smart-gaps
    #no_gaps_when_only = true
    force_split = 2
    smart_resizing = true
    smart_split = true
}

master {
    # https://wiki.hyprland.org/Configuring/Master-Layout/
    allow_small_split   = false
    special_scale_factor = 1.0
    mfact  = 0.55
    new_status = inherit
                # master slave inherit
    new_on_top  = true
    new_on_active  = before
    orientation = left
                # left right top bottom center
    inherit_fullscreen = true
    always_center_master = false
    smart_resizing = true
    drop_at_cursor = true
}

plugin = $HOME/.config/hypr/plugin/libhy3.so
plugin {
  hy3 {
    # 0 = remove the nested group
    # 1 = keep the nested group
    # 2 = keep the nested group only if its parent is a tab group
    node_collapse_policy = 0

    group_inset = 0 # offset from group split direction when only one window is in a group
    tab_first_window = false
    no_gaps_when_only = true

    tabs {
      height = 6
      padding = 3
      from_top = true  # the tab bar should animate in/out from the top instead of below the window
      rounding = 3 # rounding of tab bar corners
      render_text = false # render the window title on the bar
      text_font = Victor Nerd Font Aligned # font to render the window title with
      text_height = 8 # height of the window title
      text_padding = 3 # left padding of the window title
      col.active = rgba(1793D1ff) # active tab bar segment color
      col.urgent = 0xffff4f4f # urgent tab bar segment color
      col.inactive = 0x80333333 # inactive tab bar segment color
      col.text.active = 0xff000000 # active tab bar text color
      col.text.urgent = 0xff000000 # urgent tab bar text color
      col.text.inactive = 0xff000000  # inactive tab bar text color
    }

    autotile {
      enable = true
      ephemeral_groups = true

      # if a window would be squished smaller than this width, a vertical split will be created
      # -1 = never automatically split vertically
      # 0 = always automatically split vertically
      # <number> = pixel height to split at
      trigger_width = 0

      # if a window would be squished smaller than this height, a horizontal split will be created
      # -1 = never automatically split horizontally
      # 0 = always automatically split horizontally
      # <number> = pixel height to split at
      trigger_height = 0

      # a space or comma separated list of workspace ids where autotile should be enabled
      # it's possible to create an exception rule by prefixing the definition with "not:"
      # workspaces = 1,2 # autotiling will only be enabled on workspaces 1 and 2
      # workspaces = not:1,2 # autotiling will be enabled on all workspaces except 1 and 2
      workspaces = all # default: all
    }
  }
}

# }}}

# Auto-start Applications {{{

# See https://wiki.hyprland.org/Configuring/Keywords/ for more
exec-once = dbus-update-activation-environment --systemd XDG_CURRENT_DESKTOP=Hyprland
exec-once = /usr/lib/polkit-gnome/polkit-gnome-authentication-agent-1
# exec-once = hyprpm reload -n
exec-once = rm -rf "$HOME/.cache/" ; mkdir -p "$HOME/.cache"
exec-once = foot --server
exec-once = waybar
exec-once = gammastep
exec-once = nm-applet --indicator
exec-once = blueman-applet
exec-once = swaybg-random
exec-once = playerctld daemon
exec-once = swaync
exec-once = libinput-gestures
exec-once = dotoold
exec-once = swayosd-server

exec-once = pactl set-source-mute $(pactl --format=json list sources | python -c 'import sys,json; J=json.load(sys.stdin); print(next(s for s in J if s["description"]=="Built-in Audio Analog Stereo")["name"])') yes

# }}}

# Move focus / windows / workspace {{{
bind = SUPER, left,  hy3:movefocus, left,  visible
bind = SUPER, right, hy3:movefocus, right, visible
bind = SUPER, up,    hy3:movefocus, up,    visible
bind = SUPER, down,  hy3:movefocus, down,  visible
bind = SUPER CTRL, left,  hy3:movefocus, left
bind = SUPER CTRL, right, hy3:movefocus, right
bind = SUPER CTRL, up,    hy3:movefocus, up
bind = SUPER CTRL, down,  hy3:movefocus, down
bind = SUPER SHIFT, left,  hy3:movewindow, left
bind = SUPER SHIFT, right, hy3:movewindow, right
bind = SUPER SHIFT, up,    hy3:movewindow, up
bind = SUPER SHIFT, down,  hy3:movewindow, down

### FOR DWINDLE TESTING {{{
 ###bind = SUPER, left,  movefocus, l
 ###bind = SUPER, right, movefocus, r
 ###bind = SUPER, up,    movefocus, u
 ###bind = SUPER, down,  movefocus, d
 ###bind = SUPER SHIFT, left,  movewindow, l
 ###bind = SUPER SHIFT, right, movewindow, r
 ###bind = SUPER SHIFT, up,    movewindow, u
 ###bind = SUPER SHIFT, down,  movewindow, d
 ###bind = SUPER SHIFT, Q, killactive,
### }}}

### Master layout keybinds {{{
 bind = SUPER, left,        layoutmsg, cycleprev
 bind = SUPER, right,       layoutmsg, cyclenext
 bind = SUPER, up,          layoutmsg, cycleprev
 bind = SUPER, down,        layoutmsg, cyclenext
 bind = SUPER SHIFT, left,  layoutmsg, swapprev
 bind = SUPER SHIFT, right, layoutmsg, swapnext
#  bind = SUPER SHIFT, up,    movewindow, u
#  bind = SUPER SHIFT, down,  movewindow, d
 bind = SUPER SHIFT, Q, killactive,
### }}}

bind = SUPER SHIFT, left,  exec, [floating:1] hyprctl dispatch moveactive -10 0
bind = SUPER SHIFT, left,  exec, [floating:0] hyprctl dispatch hy3:movewindow left


bind = SUPER, E, togglesplit,
bind = SUPER, V, hy3:makegroup, v, ephemeral
bind = SUPER SHIFT, V, hy3:makegroup, h, ephemeral
bind = SUPER, backslash, hy3:makegroup, v, ephemeral
bind = SUPER SHIFT, backslash, hy3:makegroup, h, ephemeral
bind = SUPER, W, hy3:changegroup, toggletab, ephemeral
bind = SUPER SHIFT, W, hy3:makegroup, tab, ephemeral
bind = SUPER, A, hy3:changefocus, raise
bind = SUPER SHIFT, A, hy3:changefocus, lower
bind = SUPER SHIFT, Q, hy3:killactive
bind = SUPER, X, hy3:expand, expand
bind = SUPER SHIFT, X, hy3:expand, base

bindm = SUPER, mouse:272, movewindow
bindm = SUPER, mouse:273, resizewindow
bindm = SUPER CTRL, mouse:273, resizewindow 1
bindn = , mouse:272, hy3:focustab, mouse
bind = , mouse:276, exec, dotoolc <<<'key ctrl+tab'
bind = , mouse:275, exec, dotoolc <<<'key ctrl+shift+tab'


bind = SUPER, tab, exec, hyprctl --batch 'dispatch cyclenext ; dispatch bringactivetotop'
bind = SUPER, grave, workspace, empty
bind = SUPER CTRL, grave, focusurgentorlast
bind = SUPER,  space, focuswindow, floating

bind = SUPER, home, workspace, e-1
bind = SUPER, end,  workspace, e+1
bind = SUPER, 1, workspace, 1
bind = SUPER, 2, workspace, 2
bind = SUPER, 3, workspace, 3
bind = SUPER, 4, workspace, 4
bind = SUPER, 5, workspace, 5
bind = SUPER, 6, workspace, 6
bind = SUPER, 7, workspace, 7
bind = SUPER, 8, workspace, 8
bind = SUPER, 9, workspace, 9
bind = SUPER, 0, workspace, 10

bind = SUPER SHIFT, 1, movetoworkspace, 1
bind = SUPER SHIFT, 2, movetoworkspace, 2
bind = SUPER SHIFT, 3, movetoworkspace, 3
bind = SUPER SHIFT, 4, movetoworkspace, 4
bind = SUPER SHIFT, 5, movetoworkspace, 5
bind = SUPER SHIFT, 6, movetoworkspace, 6
bind = SUPER SHIFT, 7, movetoworkspace, 7
bind = SUPER SHIFT, 8, movetoworkspace, 8
bind = SUPER SHIFT, 9, movetoworkspace, 9
bind = SUPER SHIFT, 0, movetoworkspace, 10

bind = SUPER CTRL, 1, exec, hyprctl --batch 'dispatch movetoworkspacesilent 1 ; dispatch cyclenext'
bind = SUPER CTRL, 2, exec, hyprctl --batch 'dispatch movetoworkspacesilent 2 ; dispatch cyclenext'
bind = SUPER CTRL, 3, exec, hyprctl --batch 'dispatch movetoworkspacesilent 3 ; dispatch cyclenext'
bind = SUPER CTRL, 4, exec, hyprctl --batch 'dispatch movetoworkspacesilent 4 ; dispatch cyclenext'
bind = SUPER CTRL, 5, exec, hyprctl --batch 'dispatch movetoworkspacesilent 5 ; dispatch cyclenext'
bind = SUPER CTRL, 6, exec, hyprctl --batch 'dispatch movetoworkspacesilent 6 ; dispatch cyclenext'
bind = SUPER CTRL, 7, exec, hyprctl --batch 'dispatch movetoworkspacesilent 7 ; dispatch cyclenext'
bind = SUPER CTRL, 8, exec, hyprctl --batch 'dispatch movetoworkspacesilent 8 ; dispatch cyclenext'
bind = SUPER CTRL, 9, exec, hyprctl --batch 'dispatch movetoworkspacesilent 9 ; dispatch cyclenext'
bind = SUPER CTRL, 0, exec, hyprctl --batch 'dispatch movetoworkspacesilent 10 ; dispatch cyclenext'
# }}}

# Binds {{{

### l -> locked, will also work when an input inhibitor (e.g. a lockscreen) is active.
### r -> release, will trigger on release of a key.
### o -> longPress, will trigger on long press of a key.
### e -> repeat, will repeat when held.
### n -> non-consuming, key/mouse events will be passed to the active window in addition to triggering  the dispatcher.
### m -> mouse, see below.
### t -> transparent, cannot be shadowed by other binds.
### i -> ignore mods, will ignore modifiers.
### s -> separate, will arbitrarily combine keys between each mod/key
### d -> has description, will allow you to write a description for your bind.
### p -> bypasses the app's requests to inhibit keybinds.

bind = SUPER SHIFT, return, exec, terminal
#bind = SUPER SHIFT, R, forcerendererreload
bind = SUPER SHIFT, R, exec, hypr-rename-workspace
bind = SUPER SHIFT, space, togglefloating,
#bind = SUPER, D, exec, pkill wofi || wofi --show drun
bind = SUPER, D, exec, pidof wofi && pkill wofi || wofi --show drun -i -I --allow-images --width=20%
bind = SUPER SHIFT, D, exec, pidof wofi && pkill wofi || wofi --width=20% --show run,drun
bind = SUPER, P, pseudo,
bind = SUPER, F, fullscreen, 0
bind = SUPER SHIFT, F, fullscreen, 1
#bind = SUPER CTRL, F, fakefullscreen
bind = SUPER CTRL, F, fullscreenstate, 1
bind = SUPER SHIFT, S, pin, active
bind = SUPER, F8, exec, swaybg-random
bind = SUPER, C, exec, kitty --title calculator --app-id calculator -- python3 -i -c 'from math import *;import sys;sys.ps1="> "'

# Screenshot
bind = , Print, exec, XDG_CURRENT_DESKTOP=sway flameshot gui
bind = CTRL, Print, exec, grim -g "$(slurp)" - | ksnip -
bind = SUPER, Print, exec, grim -g "$(g=($(hyprctl activewindow | grep -e at: -e size: | grep -oP [0-9]+ | paste -s -)) && echo "${g[0]},${g[1]} ${g[2]}x${g[3]}")" - | ksnip -
bind = $HYPER, equal, exec, hypr-magnify


### Screen recording
bind = CTRL $HYPER, Print, exec, screen-record region
bind = $HYPER, Print, exec, screen-record fullscreen
#bindsym $hyper+$super+Print exec screen-record focused #TO-DO screen-record focused

### Brightness / Theme
bindle = ,      XF86MonBrightnessUp,   exec, swayosd-client --brightness=+10
bindle = ,      XF86MonBrightnessDown, exec, swayosd-client --brightness=-10
bindl = SHIFT,  XF86MonBrightnessUp,   exec, swayosd-client --brightness=+100
bindl = SHIFT,  XF86MonBrightnessDown, exec, swayosd-client --brightness=-99
bind = $HYPER,  XF86MonBrightnessUp,   exec, /opt/theme/theme light
bind = $HYPER,  XF86MonBrightnessDown, exec, /opt/theme/theme dark

### Turn on/off screen
bind = $HYPER, F9, exec, hyprctl monitors | grep 'dpmsStatus: 1' >/dev/null && hyprctl dispatch dpms off || hyprctl dispatch dpms on
bind = CTRL $HYPER, F9, exec, sleep 0.5 && hyprctl dispatch dpms on

### Audio and media controls
#bindle =            , XF86AudioRaiseVolume, exec, volume-adjust 5%+
#bindle =            , XF86AudioLowerVolume, exec, volume-adjust 5%-
#bindle = SHIFT      , XF86AudioRaiseVolume, exec, wpctl set-volume @DEFAULT_AUDIO_SINK@ 1%+
#bindle = SHIFT      , XF86AudioLowerVolume, exec, wpctl set-volume @DEFAULT_AUDIO_SINK@ 1%-
#bindle =            , XF86AudioMute,        exec, wpctl set-mute @DEFAULT_AUDIO_SINK@ toggle
bindle =, XF86AudioMute,        exec, swayosd-client --max-volume 120 --output-volume mute-toggle
bindle =, XF86AudioRaiseVolume, exec, swayosd-client --max-volume 120 --output-volume=5
bindle =, XF86AudioLowerVolume, exec, swayosd-client --max-volume 120 --output-volume=-5

bind   = $HYPER     , XF86AudioRaiseVolume, exec, pavucontrol
bind   = $HYPER     , XF86AudioLowerVolume, exec, pavucontrol
bind   = $HYPER     , XF86AudioMute,        exec, pavucontrol
bindle =            , XF86AudioPlay,        exec, playerctl play-pause
bindle = SHIFT      , XF86AudioPlay,        exec, playerctl-pause-next
#bindle = CTRL SHIFT , XF86AudioPlay,        exec, playerctl previous
bindle =            , XF86AudioNext,        exec, playerctl next
bindle = SHIFT      , XF86AudioNext,        exec, playerctl position 15+
bindle =            , XF86AudioPrev,        exec, playerctl previous
bindle = SHIFT      , XF86AudioPrev,        exec, playerctl position 15-
bindle = SUPER, F12, exec, playerctl next
bindle = SUPER SHIFT, F12, exec, playerctl position 15+
bindle = SUPER, F11, exec, playerctl previous
bindle = SUPER SHIFT, F11, exec, playerctl position 15-

### Launch applications
bind = $HYPER, F, exec, thunar $HOME/Desktop
bind = $HYPER SHIFT, F, exec, thunar $HOME/Desktop && hyprctl dispatch togglefloating
bind = $HYPER, B, exec, blueman-manager
bind = $HYPER, P, exec, pamac-manager
bind = $HYPER, N, exec, swaync-client --toggle-panel
bind = $HYPER, Space, exec, swaync-client --close-all ; swaync-client --open-panel ; swaync-client --close-panel
bind = $HYPER, F12, exec, pass-wofi
bind = $HYPER SHIFT, F12, exec, pass-wofi copy
bind = $HYPER, E, exec, smile


# Resize mode/submap
bind=SUPER,R,submap,resize
submap=resize
    unbind = ,down
    binde = , right, resizeactive,  100 0
    binde = , left,  resizeactive, -100 0
    binde = , down,  resizeactive,  0  -100
    binde = , up,    resizeactive,  0   100
    binde = SHIFT, right, resizeactive,  10 0
    binde = SHIFT, left,  resizeactive, -10 0
    binde = SHIFT, down,  resizeactive,  0 -10
    binde = SHIFT, up,    resizeactive,  0  10
    bind = , escape,submap,reset
    bind = , return,submap,reset
submap=reset

# System power menu
$system_state_submap = (l)ogout (r)eboot (s)leep (S)hutdown
bind=SUPER, escape, submap, $system_state_submap
submap=$system_state_submap
    bind = , l, exit
    bind = , r, exec, sudo reboot
    bind = , s, exec, sudo systemctl suspend
    bind = SHIFT, s, exec, sudo shutdown now
    bind = , escape,submap,reset
    bind = , return,submap,reset
submap=reset

# Gaps change keybinds + mode
binde = SUPER, hyphen, togglesplit
binde = SUPER, equal, exec, bash -c "hyprctl keyword general:gaps_in $(( $(hyprctl getoption general:gaps_in | grep -Po 'custom type: \K\d+') + 3 ))"
binde = SUPER_SHIFT, equal, exec, bash -c "hyprctl keyword general:gaps_out $(( $(hyprctl getoption general:gaps_out | grep -Po 'custom type: \K\d+') + 3 ))"
binde = SUPER, minus, exec, bash -c "hyprctl keyword general:gaps_in $(( $(hyprctl getoption general:gaps_in | grep -Po 'custom type: \K\d+') - 3 ))"
binde = SUPER_SHIFT, minus, exec, bash -c "hyprctl keyword general:gaps_out $(( $(hyprctl getoption general:gaps_out | grep -Po 'custom type: \K\d+') - 3 ))"

# }}}
======Config-End========

```

</details>

### Description

Since v0.45.0, it seems workspace swiping gestures have been broken. Swiping right works after a keyboard-initiated workspace change but the workspaces don't render their windows. Swiping left almost works, but swiping left twice seems to alternate back and forth. Swiping in either direction periodically crashes the Hyprland session. See my attached video below for a demo. 

I didn't see any existing issues describing this issue. I tried searching in the discord server as well, and only found this chat message which seems loosely related but I get a slightly different error message in the logs: https://discord.com/channels/961691461554950145/961694014615208048/1310304294464917645

Specifically, I get:
```
[CRITICAL] 
==========================================================================================
ASSERTION FAILED! 

Tried to render rect with width/height < 0!

at: line 2246 in OpenGL.cpp
```
whereas this other user got:
> Tried to render **shadow** with width/height < 0

Full log attached below.

I use the `hy3` plugin and layout mostly but tried disabling it and using the dwindle layout—this crashing bug was still present so I believe plugins are unrelated.

### How to reproduce

Basically just enable workspace swipe gestures, e.g. with this config block:

```makefile
gestures {
    workspace_swipe = true
    workspace_swipe_create_new = false
    workspace_swipe_fingers = 4
    workspace_swipe_invert = false
}
```

and then try swiping left and right with 4 fingers, as shown in the video below.

### Crash reports, logs, images, videos

No crash report seemed to be generated in ~/.cache after these crashes (I briefly disabled the `rm -rf ~/.cache` exec-once rule in my config above to make sure it wasn't getting deleted).

[hyprland_crash.log](https://github.com/user-attachments/files/18277465/hyprland_crash.log)

https://github.com/user-attachments/assets/211548d9-2c09-4a63-a1e8-d3dfd9016a28

**Comment by vaxerski** [comment]
does it happen with the default swipe settings? I use swipe all the time and its fine

**Comment by m-bartlett** [comment]
Seems so, tried just now with just:
```
gestures {
    workspace_swipe = true
}
```
And it crashed on the first try. 

Not sure if it's normal or useful info, but with each crash it seems to skip cleaning up the previous session:
```
/run/user/1000/hypr/788ae588979c2a1ff8a660f16e3c502ef5796755_1735445593_1261915126:
hyprland.lock  hyprland.log

/run/user/1000/hypr/788ae588979c2a1ff8a660f16e3c502ef5796755_1735503969_1958270537:
hyprland.log

/run/user/1000/hypr/788ae588979c2a1ff8a660f16e3c502ef5796755_1735504080_1544560518:
hyprland.lock  hyprland.log

/run/user/1000/hypr/788ae588979c2a1ff8a660f16e3c502ef5796755_1735588773_2047647048:
hyprland.lock  hyprland.log

/run/user/1000/hypr/788ae588979c2a1ff8a660f16e3c502ef5796755_1735588890_505015975:
hyprland.lock  hyprland.log

/run/user/1000/hypr/788ae588979c2a1ff8a660f16e3c502ef5796755_1735589971_840466909:
hyprland.lock  hyprland.log
```

**Comment by vaxerski** [comment]
well thats odd, as I can't repro at all.

**Comment by m-bartlett** [comment]
I'll try to experiment with other config options to see if that affects it

**Comment by m-bartlett** [comment]
Some progress: I copied the sample config in /usr/share/hypr/hyprland.conf to temporarily overwrite my user config to begin as a fresh control to compare against, and then copied over my regular gesture config block, the swiping worked perfectly and error free.

I also checked if `libinput-gestures` might be causing conflicts, but it also worked perfectly with the slightly modified example config and none of the above bugs were present, and all gestures and swiping worked as expected.

I'll continue to dissect which regions of my config might be causing the above bugs. Considering my and the other discord user's error logs mentioned graphics/shadows, I'll focus on config items that seem related to that.

**Comment by m-bartlett** [comment]
After some process of elimination, the source of the bug seems to be this input config for my touchpad:

```diff
device {
    name = dll0945:00-04f3:311c-touchpad
    sensitivity = 1.0 # -1.0 - 1.0, 0 means no modification.
+   accel_profile = custom 0.9, 0.0, 1.0
    #accel_profile = flat
}
```

If I have any `accel_profile` option set (including "flat"), the above bugs with swiping are present. When I comment out or remove the `accel_profile` options, the swiping works as expected.

Workspace swiping is working as I type, and my current config shows:
```hcl
device {
    name = dll0945:00-04f3:311c-touchpad
    sensitivity = 1.0 # -1.0 - 1.0, 0 means no modification.
    #accel_profile = custom 0.9, 0.0, 1.0
    #accel_profile = flat
}
```

Based on the behavior of the workspace bugs, my guess is somehow the accel_profile is causing the swipe animation to "overshoot" the workspace geometry? Just a guess.

While my muscle memory is used to the cursor movement of my custom accel_profile, I can live without it for the time being as I like the workspace swiping feature more.

**Comment by m-bartlett** [comment]
Another finding, I was digging into the blame between v0.45.0 and the latest release and didn't see any changes to `Swipe.cpp` directly mentioning acceleration, however I had a hunch and checked `libinput` releases.

The latest version of libinput `1.27.0` was around the same week as the release of Hyprland v0.45.0. I downgraded libinput to the previous release `1.26.2` and rebooted, and now my workspace swipe gestures work even with a custom `accel_profile` set .

```
$ pacman -Qn | grep libinput
libinput 1.26.2-1
```

```makefile
device {
    name = dll0945:00-04f3:311c-touchpad
    sensitivity = 1.0
    accel_profile = custom 1.0, 0.0, 1.4
}
```

Not sure yet if this bug is caused by an API change or something else, I'll investigate the diff in libinput. I'll also open an issue on the libinput repo regarding this if it seems like a bug.

**Comment by vaxerski** [comment]
hm that's odd indeed that downgrading libinput fixes this. Might be worth a shot to ask libinput guys what they changed. Could also bisect libinput, that would help them.

**Comment by m-bartlett** [comment]
Well, while performing the bisect I first just tried building libinput from the latest commit and the swiping bug is gone. Based on the history since that release tag, they made some bugfixes to the swipe gesture that might explain it:

```console
$ git log --oneline 1.27.0..HEAD | grep -i gesture
02887813 gestures: change the debug log messages for state debugging
3563b678 gestures: use a macro for debugging transition states
ce7b7c94 gestures: fix transformation of scroll to pinch
c3aa00ef gestures: don't handle SWIPE if we transitioned from SWIPE_START~/Desktop
```

Future releases of libinput will likely contain the fix, so I'll close this issue
